### PR TITLE
gh-2580 Replace traditional security with OAuth2/OpenIDConnect

### PR DESCRIPTION
**Pull request by ghillert**
* Remove the option to disable Authorization
* Remove `ApacheDSContainerWithSecurity`  …
* Remove LDAP related code
* Remove File Based Authentication
* Remove traditional security option

resolves https://github.com/spring-cloud/spring-cloud-dataflow/issues/2580
resolves https://github.com/spring-cloud/spring-cloud-dataflow/issues/2579

depends on https://github.com/spring-cloud/spring-cloud-common-security-config/pull/24

depends on https://github.com/spring-cloud/spring-cloud-common-security-config/pull/28

### CODE REVIEW

File: spring-cloud-dataflow-server-local/src/main/resources/application.yml [hunk 1]
```diff
@@ -8,8 +8,14 @@ spring:
     deployer:
       local:
         free-disk-space-percentage: 0
+#    dataflow:
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-server-local/src/main/resources/application.yml]: We can probably delete this.

File: spring-cloud-dataflow-server-local/src/main/resources/application.yml [hunk 2]
```diff
@@ -43,6 +38,18 @@ logging:
 #    resource:
 #      user-info-uri: http://127.0.0.1:9999/me
 
+# UAA Security
+
+security:
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-server-local/src/main/resources/application.yml]: Maybe we could just document the appropriate values for the OAuth2 properties and delete the following entries (with random default values) from local server application config.

File: spring-cloud-starter-dataflow-server-local/src/test/java/org/springframework/cloud/dataflow/server/local/security/LocalServerSecurityRootWithUsersFileTests.java [hunk 3]
```diff
@@ -49,6 +50,7 @@
 	private static UserCredentials fullUser = new UserCredentials("fulluser", "fullpassword");
 
 	@Test
+	@Ignore("Revisit once Outh2/OpenID Connect security is implemented")
```
Comment (ilayaperumalg) [inline spring-cloud-starter-dataflow-server-local/src/test/java/org/springframework/cloud/dataflow/server/local/security/LocalServerSecurityRootWithUsersFileTests.java]: Was this waiting on https://github.com/spring-cloud/spring-cloud-common-security-config/commit/cb190c3c80b4904090b13224e419e17d8e1238cb ?

**Comment by jvalkeal** [comment]
Btw this PR needs a rebase

**Comment by ghillert** [comment]
Rebased. Make sure this works with Spring Boot 2.1 - THX to @jvalkeal !

**Comment by jvalkeal** [comment]
lgtm, gonna remove those debug logs changes in local server when merging.

**Comment by jvalkeal** [comment]
merged per 782eb82940417943aa655e4e19eca895541d748d

**Pull request closed without merging**